### PR TITLE
feat(functions): add jaro_winkler string similarity function

### PR DESCRIPTION
**Pull request by maxjustus**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

Adds a jaro_winkler string edit distance function taken from this project https://github.com/joshuaclayton/jaro_winkler

## Tests

- [x] Unit Test
- [x] Logic Test
- [ ] Benchmark Test
- [ ] No Test - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [x] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [ ] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/databendlabs/databend/16993)

### CODE REVIEW

File: src/query/functions/src/scalars/string.rs [hunk 1]
```diff
@@ -841,6 +854,190 @@ pub(crate) mod soundex {
     }
 }
 
+// this implementation comes from https://github.com/joshuaclayton/jaro_winkler
+pub(crate) mod jaro_winkler {
```
Comment (TCeason) [inline src/query/functions/src/scalars/string.rs:858]: The function jaro_winkler and mod jaro_winkler can refactor to file other.rs.

File: src/query/functions/src/scalars/string.rs [hunk 2]
```diff
@@ -841,6 +854,190 @@ pub(crate) mod soundex {
...
+        ((max / 2.0).floor() - 1.0) as usize
+    }
+
+#[cfg(test)]
+    mod tests {
+        use super::*;
+
+        #[test]
+        fn different_is_zero() {
+            assert_eq!(jaro_winkler("foo", "bar"), 0.0);
```
Comment (TCeason) [inline src/query/functions/src/scalars/string.rs:1002]: Please impl the ut in https://github.com/datafuselabs/databend/blob/629ee7bab86bf74538814e794122e1fedbbeb288/src/query/functions/tests/it/scalars/other.rs

**Comment by maxjustus** [comment]
ok @TCeason - I've moved the definition to other.rs

**Comment by TCeason** [comment]
> ok @TCeason - I've moved the definition to other.rs

Please fmt code

**Comment by maxjustus** [comment]
ok done @TCeason !

**Comment by TCeason** [comment]
> ok done @TCeason !

Cool. But ci check failed.

**Review by TCeason — APPROVED** [review]
Thank you

**Pull request merged**